ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Protocol/Praos.hs

```diff
@@ -63,14 +65,17 @@ instance PraosCrypto c => ProtocolHeaderSupportsEnvelope (Praos c) where
 
   type EnvelopeCheckError _ = PraosEnvelopeError
 
-  envelopeChecks cfg lv hdr = do
+  envelopeChecks cfg checkpoints lv hdr = do
     unless (m <= maxpv) $ throwError (ObsoleteNode m maxpv)
     unless (fromIntegral (bhviewHSize bhv) <= maxHeaderSize) $
       throwError $
         HeaderSizeTooLarge (fromIntegral $ bhviewHSize bhv) maxHeaderSize
     unless (bhviewBSize bhv <= maxBodySize) $
       throwError $
         BlockSizeTooLarge (bhviewBSize bhv) maxBodySize
+    whenJust (Map.lookup (pHeaderBlock hdr) checkpoints) $ \checkpoint ->
+      unless (checkpoint == pHeaderHash hdr) $
```

Comment:
same comment as other guard

Reply:
Yeah, I am always paralyzed by the choice between `when` and `unless`... I used `unless` here as it is also used in all preceeding checks (even though it would be easy to rewrite these places from eg `unless (m <= maxpv)` to `when (m > maxpv)`), and then also used it in all other `additionalEnvelopeChecks` for consistency.

Reply:
Switched unless to when in 5c93fa0